Harmony/MethodPatcher.cs

```diff
@@ -328,6 +328,8 @@ static void EmitCallParameter(ILGenerator il, MethodBase original, MethodInfo pa
 				{
 					if (AccessTools.GetReturnedType(original) == typeof(void))
 						throw new Exception("Cannot get result from void method " + original.FullDescription());
+					if (AccessTools.GetReturnedType(original) != patchParam.ParameterType)
```

Comment:
I don't think types should be compared with == here. What is needed here is a check for assignable since the patch could have a superclass or even `object` as a return type.